From 7227878d9f64311c447bc7eca2a7eef90b8f57fc Mon Sep 17 00:00:00 2001
From: Gilberto Garcia <gilbertodamian14@gmail.com>
Date: Sun, 21 Apr 2019 18:00:57 -0500
Subject: [PATCH] Modified Injector creation point Now the Guice Injector is
 created before the plugins load, not after that


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index a573b556..12c985db 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -261,7 +261,8 @@ public final class CraftServer implements Server {
         // enablePlugins(PluginLoadOrder.STARTUP);
         // Spigot End
 
-        injector = Guice.createInjector(Stage.PRODUCTION, new ServerModule(), new CraftModule(this, getPluginManager(), getScheduler(), getServicesManager()), new PluginModule(this.getPluginManager().getPlugins()));
+        // Now the injector is created before the plugins load, not after that
+        // injector = Guice.createInjector(Stage.PRODUCTION, new ServerModule(), new CraftModule(this, getPluginManager(), getScheduler(), getServicesManager()), new PluginModule(this.getPluginManager().getPlugins()));
     }
 
     public boolean getCommandBlockOverride(String command) {
@@ -293,6 +294,9 @@ public final class CraftServer implements Server {
     }
 
     public void loadPlugins() {
+		// Here we create the Guice injector
+        injector = Guice.createInjector(Stage.PRODUCTION, new ServerModule(), new CraftModule(this, getPluginManager(), getScheduler(), getServicesManager()), new PluginModule(this.getPluginManager().getPlugins()));
+
         pluginManager.registerInterface(JavaPluginLoader.class);
 
         File pluginFolder = (File) console.options.valueOf("plugins");
-- 
2.21.0.windows.1

