From 7307f1893627a035aa5a43bfc5c614bfc43b4b2c Mon Sep 17 00:00:00 2001
From: TomateDeveloper <ianfe@martina.com.co>
Date: Thu, 9 Jan 2020 17:33:01 -0500
Subject: [PATCH] Added functions and updated player base volume


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 4759ddc92..4f38eb6c0 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -61,9 +61,9 @@ import java.util.logging.Logger;
 @DelegateDeserialization(CraftOfflinePlayer.class)
 public class CraftPlayer extends CraftHumanEntity implements Player {
 
-    private byte playerVolume = 100;
+    private byte playerVolume = 50;
     private Set<Hologram> holograms = new HashSet<>();
-    private List<SongPlayer> songPlayer = new ArrayList<>();
+    @Nullable private SongPlayer personalRadio;
     private String databaseIdentifier = "";
     private long firstPlayed = 0;
     private long lastPlayed = 0;
@@ -83,7 +83,6 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     public CraftPlayer(CraftServer server, EntityPlayer entity) {
         super(server, entity);
-
         firstPlayed = System.currentTimeMillis();
     }
 
@@ -1497,6 +1496,16 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         if (this.databaseIdentifier.equalsIgnoreCase("")) this.databaseIdentifier = identifier;
     }
 
+    @Override
+    public @Nullable SongPlayer getPersonalRadio() {
+        return this.personalRadio;
+    }
+
+    @Override
+    public void shareRadio(Player player) {
+        if (this.personalRadio != null) this.personalRadio.addPlayer(player);
+    }
+
     @Override
     public byte getPlayerVolume() {
         return this.playerVolume;
@@ -1509,24 +1518,26 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     @Override
     public void stopPlayingSong() {
-        this.songPlayer.get(0).removePlayer(this);
-        this.songPlayer.clear();
+        if (this.personalRadio != null) {
+            this.personalRadio.setPlaying(false);
+            this.personalRadio = null;
+        }
     }
 
     @Override
     public void playSong(String fileName) {
-        File songFile = new File("song/" + fileName + ".nbs");
+        File songFile = new File("songs/" + fileName + ".nbs");
         if (!songFile.exists() || songFile.isDirectory()) throw new IllegalArgumentException("The song file could not be found");
         Song song = NBSDecoder.parse(songFile);
         RadioSongPlayer radioPlayer = new RadioSongPlayer(song);
         radioPlayer.addPlayer(this);
-        this.songPlayer.clear();
-        this.songPlayer.add(radioPlayer);
+        radioPlayer.setPlaying(true);
+        this.personalRadio = radioPlayer;
     }
 
     @Override
     public boolean isReceivingSong() {
-        return this.songPlayer.size() >= 1;
+        return this.personalRadio != null;
     }
 
     @Override
-- 
2.22.0.windows.1

