From b21b0547933ac712b2d7b9e15013e3e3bf87de10 Mon Sep 17 00:00:00 2001
From: yusshu <yushu@unnamed.team>
Date: Fri, 4 Jun 2021 23:50:57 -0500
Subject: [PATCH] Make CraftObjective class public


diff --git a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
index 5bd8cf9d..7e604fb3 100644
--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
+++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftObjective.java
@@ -9,17 +9,17 @@ import org.bukkit.scoreboard.DisplaySlot;
 import org.bukkit.scoreboard.Objective;
 import org.bukkit.scoreboard.Score;
 
-final class CraftObjective extends CraftScoreboardComponent implements Objective {
+public final class CraftObjective extends CraftScoreboardComponent implements Objective {
     private final ScoreboardObjective objective;
     private final CraftCriteria criteria;
 
-    CraftObjective(CraftScoreboard scoreboard, ScoreboardObjective objective) {
+    public CraftObjective(CraftScoreboard scoreboard, ScoreboardObjective objective) {
         super(scoreboard);
         this.objective = objective;
         this.criteria = CraftCriteria.getFromNMS(objective);
     }
 
-    ScoreboardObjective getHandle() {
+    public ScoreboardObjective getHandle() {
         return objective;
     }
 
-- 
2.31.1

