From daacb65168d34a05d0eea0f47ba76ac96efac5ac Mon Sep 17 00:00:00 2001
From: Gilberto Garcia <gilbertodamian14@gmail.com>
Date: Sat, 20 Apr 2019 23:34:16 -0500
Subject: [PATCH] We need to create this method Bukkit unit testing complains
 if the method doesn't exist in Server and Bukkit


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 8fbc8283..65af6214 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -12,6 +12,7 @@ import java.util.Set;
 import java.util.UUID;
 import java.util.logging.Logger;
 
+import com.google.inject.Injector;
 import org.bukkit.Warning.WarningState;
 import org.bukkit.command.CommandException;
 import org.bukkit.command.CommandMap;
@@ -435,6 +436,10 @@ public final class Bukkit {
         return server.getPlayer(id);
     }
 
+    public static Injector getInjector() {
+        return server.getInjector();
+    }
+
     /**
      * Gets the plugin manager for interfacing with plugins.
      *
-- 
2.21.0.windows.1

