From 8a513766d4fb68ada953616bd9a6894ff361da40 Mon Sep 17 00:00:00 2001
From: TomateDeveloper <ianfe@martina.com.co>
Date: Tue, 14 Apr 2020 21:51:03 -0500
Subject: [PATCH] Updated line processor


diff --git a/src/main/java/org/bukkit/scoreboard/BoardEntry.java b/src/main/java/org/bukkit/scoreboard/BoardEntry.java
new file mode 100644
index 00000000..b5f6b409
--- /dev/null
+++ b/src/main/java/org/bukkit/scoreboard/BoardEntry.java
@@ -0,0 +1,9 @@
+package org.bukkit.scoreboard;
+
+public interface BoardEntry {
+
+    String getPrefix();
+
+    String getSuffix();
+
+}
diff --git a/src/main/java/org/bukkit/scoreboard/BoardLine.java b/src/main/java/org/bukkit/scoreboard/BoardLine.java
index 057825e3..b68a489a 100644
--- a/src/main/java/org/bukkit/scoreboard/BoardLine.java
+++ b/src/main/java/org/bukkit/scoreboard/BoardLine.java
@@ -2,7 +2,7 @@ package org.bukkit.scoreboard;
 
 public interface BoardLine {
 
-    String getText();
+    BoardEntry getText();
 
     int getPosition();
 
-- 
2.22.0.windows.1

